griptape/engines/rag/modules/query/translate_query_rag_module.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+from attrs import Factory, define, field
+
+from griptape.engines.rag.modules import BaseQueryRagModule
+from griptape.utils import J2
+
+if TYPE_CHECKING:
+    from griptape.drivers import BasePromptDriver
+    from griptape.engines.rag import RagContext
+
+
+@define(kw_only=True)
+class TranslateQueryRagModule(BaseQueryRagModule):
+    prompt_driver: BasePromptDriver = field()
+    language: str = field()
+    generate_user_template: Callable[[str, str], str] = field(
+        default=Factory(lambda self: self.default_user_template_generator, takes_self=True),
+    )
```

Comment:
Should this be a `J2` object for consistency with [other places](https://github.com/griptape-ai/griptape/blob/a20309ea63d08c140dce2bd449ea56af4f086ed7/griptape/engines/summary/prompt_summary_engine.py#L23-L24)?

Reply:
We should actually move `SummaryConversationMemory` and other static templates to be `Callables` so that the user has more control over how the template is generated. Similar to what we do in `PromptTask` and `ToolkitTask`.

Reply:
Fair point, though the type signature of `Callable[[str, str], str]` might be confusing for users to interact with. Maybe we should do `Callable[dict], str`?

Reply:
`dict` doesn't make sense because I only pass query and language to the method.